lib/board.rb

```diff
@@ -0,0 +1,51 @@
+require 'pry'
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    game_board = @cells.map{|c| " #{c} "}
+    puts game_board[0, 3].join("|")
+    puts "-----------"
+    puts game_board[3, 3].join("|")
+    puts "-----------"
+    puts game_board[6, 3].join("|")
+  end
+
+  def position(input)
+    @cells[input.to_i - 1]
+  end
+
+  def full?
+    @cells.all? {|c| c == "X" || c == "O"}
+  end
+
+  def turn_count
+    @cells.count{|c| c == "X" || c == "O"}
+  end
+
+  def taken?(input)
+    cell = position(input)
+    cell == "X" || cell == "O" ? true : false
+  end
+
+  def valid_move?(input)
+    i = input.to_i
+    i <= 9 && i > 0 && !taken?(input)
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`position.between?(1,9)`